src/main/java/com/mongodb/jdbc/MongoConnection.java

```diff
@@ -190,6 +197,33 @@ protected void checkConnection() throws SQLException {
         }
     }
 
+    private MongoClusterType determineClusterType() {
+        BsonDocument buildInfoCmd = new BsonDocument();
+        buildInfoCmd.put("buildInfo", new BsonInt32(1));
+
+        // The { buildInfo: 1 } command returns information that indicates
+        // the type of the cluster.
+        BuildInfo buildInfoRes =
+                mongoClient
+                        .getDatabase("admin")
+                        .withCodecRegistry(MongoDriver.registry)
+                        .runCommand(buildInfoCmd, BuildInfo.class);
+
+        // If the "dataLake" field is present, it must be an ADF cluster.
+        if (buildInfoRes.dataLake != null) {
```

Comment:
I was thinking this is something ADF would eventually change this to the correct naming, but it looks like they thought to address it and decided not to fix it so I think we're ok here.  [MHOUSE-6693](https://jira.mongodb.org/browse/MHOUSE-6693)